src/dag.rs

```diff
@@ -27,40 +27,16 @@ use crate::{Imr, Value};
 
 /// Generic container for Simplicity DAGs
 pub enum Dag<T> {
```

Comment:
b4f11b4e205413003b6c58fb595ed57853808534: We should update this description.

All these different graph types can be confusing. `Dag` is a pretty generic term, so it `dag.rs`. Maybe we should rename it to `iter.rs` :) IDK what `Dag` would become. Happy to track this in an issue and make a PR later.

Reply:
Let's file a separate issue. I agree calling the module `iter` would make more sense. `Dag` I'm unsure of, I think the current name is ok. I wouldn't mind `DagNode` though I think that could be confused with `Node`.

Worth noting that users almost never interact with this enum directly.